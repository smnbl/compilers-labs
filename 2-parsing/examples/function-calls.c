int function()
{
    int x = foo();
    int y = bar(1);
    int z = baz(1, y);

    foobar();
}
