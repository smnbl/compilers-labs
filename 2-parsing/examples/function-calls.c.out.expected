└── Program
    └── FuncDecl: returnType = 'int', name = 'function'
        └── CompoundStmt
            ├── VarDecl: type = 'int', name = 'x'
            │   └── FuncCallExpr: name = 'foo'
            ├── VarDecl: type = 'int', name = 'y'
            │   └── FuncCallExpr: name = 'bar'
            │       └── IntLiteral: value = '1'
            ├── VarDecl: type = 'int', name = 'z'
            │   └── FuncCallExpr: name = 'baz'
            │       ├── IntLiteral: value = '1'
            │       └── VarRefExpr: name = 'y'
            └── ExprStmt
                └── FuncCallExpr: name = 'foobar'
